From 0a58da316862f23f9a3cbf474440bef4c8f8c4f0 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Wed, 9 Jan 2013 14:53:59 +0200
Subject: [PATCH 046/103] geiger-cape: Convert to using the new ADC driver

Now that the proper driver has DT binding convert to using it.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
---
 firmware/capes/cape-bone-geiger-00A0.dts | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/firmware/capes/cape-bone-geiger-00A0.dts b/firmware/capes/cape-bone-geiger-00A0.dts
index 18bf1f2..287d75c 100644
--- a/firmware/capes/cape-bone-geiger-00A0.dts
+++ b/firmware/capes/cape-bone-geiger-00A0.dts
@@ -54,6 +54,10 @@
 		target = <&ocp>;
 		__overlay__ {
 
+			/* avoid stupid warning */
+			#address-cells = <1>;
+			#size-cells = <1>;
+
 			gpio-leds-cape-geiger {
 				compatible = "gpio-leds";
 				pinctrl-names = "default";
@@ -75,9 +79,17 @@
 
 			};
 
-			tscadc-cape-geiger {
-				compatible = "ti-tscadc-dt";
-				adc-channels = <8>;
+			tscadc {
+				compatible = "ti,ti-tscadc";
+				reg = <0x44e0d000 0x1000>;
+
+				interrupt-parent = <&intc>;
+				interrupts = <16>;
+				ti,hwmods = "adc_tsc";
+
+				adc {
+					adc-channels = <8>;
+				};
 			};
 
 			bone-cape-geiger {
-- 
1.8.1.4

